tests/rptest/tests/redpanda_oauth_test.py

```diff
@@ -184,6 +187,80 @@ def check_sr_subjects():
 
         wait_until(check_sr_subjects, timeout_sec=5)
 
+    @cluster(num_nodes=4)
+    def test_java_client(self):
+        kc_node = self.keycloak.nodes[0]
+
+        self.keycloak.admin.create_user('norma',
+                                        'desmond',
+                                        realm_admin=True,
+                                        email='10086@sunset.blvd')
+        self.keycloak.login_admin_user(kc_node, 'norma', 'desmond')
+        self.keycloak.admin.create_client(CLIENT_ID)
+
+        # add an email address to myapp client's service user. this should
+        # appear alongside the access token.
+        self.keycloak.admin.update_user(f'service-account-{CLIENT_ID}',
+                                        email='myapp@customer.com')
+
+        service_user_id = self.keycloak.admin_ll.get_user_id(
+            f'service-account-{CLIENT_ID}')
+
+        self.rpk.create_topic(EXAMPLE_TOPIC)
+        self.rpk.sasl_allow_principal(f'User:{service_user_id}', ['all'],
+                                      'topic', EXAMPLE_TOPIC, self.su_username,
+                                      self.su_password, self.su_algorithm)
+
+        cfg = self.keycloak.generate_oauth_config(kc_node, CLIENT_ID)
+        assert cfg.client_secret is not None
+        assert cfg.token_endpoint is not None
+
+        cli = KafkaCliTools(self.redpanda, oauth_cfg=cfg)
+
+        expected_topics = set([EXAMPLE_TOPIC])
+        wait_until(lambda: set(cli.list_topics()) == expected_topics,
+                   timeout_sec=5)
+
+        self.redpanda.logger.debug(
+            "Clear the deny ACL and grant access to service user. We can produce now."
+        )
+
+        N_REC = 10
+        cli.oauth_produce(EXAMPLE_TOPIC, N_REC)
+        records = []
+        for i in range(0, N_REC):
+            rec = self.rpk.consume(EXAMPLE_TOPIC, n=1, offset=i)
+            records.append(json.loads(rec))
+
+        self.redpanda.logger.debug(json.dumps(records, indent=1))
+
+        assert len(
+            records) == N_REC, f"Expected {N_REC} records, got {len(records)}"
+
+        values = set([r['value'] for r in records])
+
+        assert len(values) == len(
+            records
+        ), f"Expected {len(records)} unique records, got {len(values)}"
+
+        self.redpanda.logger.debug(
+            "Deny access to the service user. Produce should fail with an authZ error."
+        )
```

Comment:
This makes sense, although the default is deny, so it may be simpler to attempt the deny case before calling  `self.rpk.sasl_allow_principal`

Reply:
Oh, the default is to deny? I didn't see any restriction at all by default, which is why I added this whole dance, extra bit on Rpk, etc. Might be a logic error in my setup then.

Reply:
Yeah when authz is enabled, if there's no matching "allow" acl it'll deny